Services/SSLCertificateService.cs

```diff
@@ -196,11 +198,13 @@ public X509Certificate2 GetCertificate(StoreLocation storeLocation, StoreName st
                 {
                     if (IsCertificateMatchingNotNullParameters(certThumbPrint, friendlyName, certSubjectName, dnsName, notAfter, cert))
                     {
-                        return cert;
+                        result.Add(cert);
                     }
                 }
             }
-            return null;
+            if (!result.Any())
+                return null;
+            return result.OrderByDescending(cert=>cert.NotAfter).FirstOrDefault();
```

Comment:
instead of having a list ( we're only interested in one item), why dont we do this orderby to certificates, and return the first match ? maybe is less efficient to do the order by to the whole collection? ,but on this way we're forced to proccess all the items in certificates, add them to the list and then order them, I think we should avoid this by doing what I said